inf-ruby.el

```diff
@@ -282,6 +284,20 @@ The following commands are available:
       (inf-ruby-remove-in-string (buffer-substring (point) end)
                                  inf-ruby-prompt-pattern))))
 
+(defun inf-ruby-buffer ()
+  "Return inf-ruby buffer for the current buffer or project."
+  (catch 'buffer
+    (let ((current-dir         (file-truename default-directory))
+          (current-project-dir (file-truename
+                                (locate-dominating-file default-directory
+                                                        #'inf-ruby-console-match))))
+      (dolist (buffer inf-ruby-buffers)
+        (when (buffer-live-p buffer)
+          (with-current-buffer buffer
+            (when (member (file-truename default-directory)
+                          (list current-dir current-project-dir))
```

Comment:
I think we should handle that in `ruby-switch-to-inf` (and maybe in `inf-ruby` as well; and make `run-ruby` not reuse existing buffers). Instead of `(inf-ruby-buffer)`, they will use `(or (inf-ruby-buffer) inf-ruby-buffer)`.